pkg/arvo/sys/vane/clay.hoon

```diff
@@ -2301,6 +2323,140 @@
         (emit hen %pass /what %$ what/fil)
       --
     --
+  ::  +sweep-zuse-cache: garbage collect zuse cache
+  ::
+  ++  sweep-zuse-cache
+    ^+  ..park
+    =/  using=(set weft)
+      %-  ~(rep by dos.rom)
+      |=  [[=desk =dojo] wefts=(set weft)]
+      (~(put in wefts) wef.dom.dojo)
+    =/  stale=(list weft)
+      ~(tap in (~(dif in ~(key by zuz)) using))
+    |-
+    ?~  stale
+      ..park
+    $(stale t.stale, zuz (~(del by zuz) i.stale))
+  ::
+  ::  +got-zuse: find and build zuse at kelvin
+  ::
+  ++  got-zuse
+    |=  waf=waft
+    |^  ^-  [[weft vase] _..park]
+    =/  welt=(list weft)  (sort-filter-waft waf)
+    =/  wefts=(set weft)  (silt welt)
+    ?^  cached=(try-zuse-cache welt)
+      [u.cached ..park]
+    =/  oldest=weft  (rear welt)
+    =/  =dome  dom:(~(got by dos.rom) %base)
+    =/  =aeon  let.dome
+    |-
+    ?.  (gte aeon 1)
+      (mean leaf+"clay: can't find zuse for {<(waft-to-wefts waf)>}" ~)    
+    =/  =yaki  (~(got by hut.ran) (~(got by hit.dome) aeon))
+    ?~  aeon-waft=(get-waft yaki)
+      $(aeon (dec aeon))
+    ?~  aeon-welt=(sort-filter-waft u.aeon-waft)
+      $(aeon (dec aeon))
+    =/  aeon-newest=weft  i.aeon-welt
+    :: if the newest weft for this aeon is older than
+    :: the oldest target weft, do not continue
+    ::
+    ?:  (gth num.aeon-newest num.oldest)
+      (mean leaf+"clay: gave up zuse search for {<(waft-to-wefts waf)>}" ~)    
+    ?~  (~(int in wefts) (silt aeon-welt))
+      $(aeon (dec aeon))
+    ?~  files=(get-kernel-files yaki)
+      $(aeon (dec aeon))
+    =/  zuse-core=vase  (build-zuse u.files)
+    =/  actual=weft  [%zuse !<(@ud (slap zuse-core %limb %zuse))]
+    ?.  (~(has in wefts) actual)
+      (mean leaf+"clay: zuse to /sys/kelvin mismatch at aeon {<aeon>}" ~)    
+    :-  [actual zuse-core]
+    ..park(zuz (~(put by zuz) actual zuse-core))
+    ::
+    ::  +build-zuse: compile zuse from given files
+    ::
+    ++  build-zuse
+      |=  [h=@ a=@ l=@ z=@]
+      ^-  vase
+      =/  hoon-core=vase
+        %-  (slog 'clay: compiling hoon' ~)
+        =-  (slot 7 -)
+        (road |.((slap *vase (ream h))))
```

Comment:
shouldn't this and the arvo build also use `+rain`, not `+ream`?

Reply:
arvo itself doesn't use rain for these two, I just copied that. I can change it if it makes more sense though, I was just trying to have things build identically

Reply:
ah ok -- better to replicate arvo's build process exactly, so it's goof as-is